vul-scans/action.yaml

```diff
@@ -0,0 +1,282 @@
+      if: ${{ inputs.RUN_SNYK }} == "true"
+      shell: bash
+      id: snyk-attest
+      env:
+        COSIGN_EXPERIMENTAL: "true"
+      run: |
+        ATTESTATION=$(mktemp)
+
+        cat > "${ATTESTATION}" <<EOF
+        {
+            "invocation": {
+              "parameters": null,
+              "uri": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}",
+              "event_id": "${{ github.event.id }}",
+              "builder.id": "${{ github.workflow }}"
+            },
+            "scanner": {
+              "uri": "${SNYK_URL}",
+              "version": "${SNYK_VERSION}",
+              "result": $(cat snyk.json | jq .)
+            },
+            "metadata": {
+              "scanStartedOn": "${{ steps.scan-start.outputs.date }}",
+              "scanFinishedOn": "$(TZ=Zulu date "+%Y-%m-%dT%H:%M:%SZ")"
+            }
+        }
+        EOF
+        echo "Snyk COSIGN_EXPERIMENTAL: $COSIGN_EXPERIMENTAL"
+        
+        export COSIGN_EXPERIMENTAL="$COSIGN_EXPERIMENTAL"
+        
+        cosign attest ${{ inputs.image }} \
+            --type vuln --predicate "${ATTESTATION}" \
+            --attachment-tag-prefix snyk-
+
+    - name:  Scan image with AquaSec/Trivy
+      uses: aquasecurity/trivy-action@0105373003c89c494a3f436bd5efc57f3ac1ca20 #v0.5.1
+      id: trivy-scan
+      with:
+        image-ref: ${{ inputs.image }}
+        format: 'sarif'
+        output: 'trivy-results.sarif'
+        exit-code: 0
+        vuln-type: 'os,library'
+        severity: 'CRITICAL,HIGH,MEDIUM,LOW,UNKNOWN'
+
+    - name: Cosign Attest trivy results
+      shell: bash
+      id: trivy-attest
+      env:
+        COSIGN_EXPERIMENTAL: "true"
+      run: |
+        ATTESTATION=$(mktemp)
+
+        export SCANNER_URI=$(cat trivy-results.sarif | jq .runs[0].tool.driver.informationUri)
+        export SCANNER_VERSION=$(cat trivy-results.sarif | jq .runs[0].tool.driver.version)
+
+        cat > "${ATTESTATION}" <<EOF
+        {
+            "invocation": {
+              "parameters": null,
+              "uri": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}",
+              "event_id": "${{ github.event.id }}",
+              "builder.id": "${{ github.workflow }}"
+            },
+            "scanner": {
+              "uri": ${SCANNER_URI},
+              "version": ${SCANNER_VERSION},
+              "result": $(cat trivy-results.sarif | jq .)
+            },
+            "metadata": {
+              "scanStartedOn": "${{ steps.scan-start.outputs.date }}",
+              "scanFinishedOn": "$(TZ=Zulu date "+%Y-%m-%dT%H:%M:%SZ")"
+            }
+        }
+        EOF 
+               
+        echo "trivy COSIGN_EXPERIMENTAL: $COSIGN_EXPERIMENTAL"
+        
+        export COSIGN_EXPERIMENTAL="$COSIGN_EXPERIMENTAL"
+               
+        cosign attest ${{ inputs.image }} \
+            --type vuln --predicate "${ATTESTATION}" \
+            --attachment-tag-prefix trivy-
```

Comment:
This changes the namespace of cosign attachments, we should just attach multiple attestations using the standard name instead.

Reply:
ok, well stuff like `policy-controller` doesn't expose any way to validate these, so for these to be useful in those contexts would need some work.

cc @vaikas in case he has a better sense for how we encode multiple vulnz scans, which seems like a perfectly reasonable use case

Reply:
i feel like policy-controller should expose that since it's an option in cosign.